app/models/miq_server/role_management.rb

```diff
@@ -50,10 +50,15 @@ def set_active_role_flags
     save
   end
 
-  def set_assigned_roles
+  def sync_assigned_roles
     self.role = ::Settings.server.role
   end
 
+  def ensure_default_roles
+    MiqServer.my_server.add_settings_for_resource(:server => {:role => ENV["MIQ_SERVER_DEFAULT_ROLES"]}) if role.blank? && ENV["MIQ_SERVER_DEFAULT_ROLES"].present?
```

Comment:
Ohhhh I see...if you don't change it then you end up with it syncing back from config and everything is wonky.  Even so, I don't like using `add_settings_for_resource` because it broadcasts a config change to all running servers.